lib/pf/api.pm

```diff
@@ -1360,9 +1361,50 @@ sub handle_accounting_metadata : Public {
             pf::log::get_logger->debug("Not handling iplog update because we're not configured to do so on accounting packets.");
         }
     }
+    $client->notify("firewallsso_accounting", %RAD_REQUEST);
+
     return $return;
 }
 
+=head2 firewallsso_accounting
+
+Update the firewall sso based on radius accounting
+
+=cut
+
+sub firewallsso_accounting : Public {
+    my ($class, %RAD_REQUEST) = @_;
+    my $logger = pf::log::get_logger();
+    if ($RAD_REQUEST{'Calling-Station-Id'} && $RAD_REQUEST{'Framed-IP-Address'} && pf::util::isenabled($pf::config::Config{advanced}{sso_on_accounting})) {
+        my $mac = pf::util::clean_mac($RAD_REQUEST{'Calling-Station-Id'});
+        my $node = pf::node::node_attributes($mac);
+        my $ip = $RAD_REQUEST{'Framed-IP-Address'};
+        if($ip){
+            my $firewallsso_method = "Stop";
+            my $timeout = '3600'; #Default to 1 hour
+
+            if ($node->{status} eq $pf::node::STATUS_REGISTERED) {
+                $firewallsso_method = "Update";
+                if ($node->{unregdate} ne '0000-00-00 00:00:00') {
+                    my $time = DateTime::Format::MySQL->parse_datetime($node->{unregdate});
+                    $time->set_time_zone("local");
+                    my $now = DateTime->now(time_zone => "local");
+                    $timeout = $time->epoch - $now->epoch;
+                }
+            }
+
```

Comment:
You don't seem to be checking for MAC change by an IP like in DHCP where a device would take the IP of another device because the first became inactive